### PR TITLE
CLOUDP-68637: implement e2e tests for apikeys/whitelist

### PR DESCRIPTION
**Pull request by andreaangiolillo**
## Proposed changes

_Jira ticket:_ CLOUDP-68637

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: e2e/iam/helper_test.go [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+	var key mongodbatlas.APIKey
+	if err := json.Unmarshal(resp, &key); err != nil {
+		return "", err
+	}
+
+	if key.ID != "" {
+		return key.ID, nil
+	}
+
+	return "", fmt.Errorf("the apiKey ID is empty")
```
Comment (gssbzn) [inline e2e/iam/helper_test.go:66]: [nit] don't use `fmt.Errorf` if you don't need to format the text
```suggestion
	return "", errors.New("the apiKey ID is empty")
```

File: e2e/iam/org_api_key_whitelist_test.go [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+			"-o=json")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		a := assert.New(t)
+		if a.NoError(err, string(resp)) {
+			var key mongodbatlas.WhitelistAPIKeys
+			if err := json.Unmarshal(resp, &key); err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+			a.NotEmpty(key.Results)
```
Comment (gssbzn) [inline e2e/iam/org_api_key_whitelist_test.go:59]: [nit]
```suggestion
			if err := json.Unmarshal(resp, &key); a.NoError(err) {
				a.NotEmpty(key.Results)
			}

```

File: e2e/iam/helper_test.go [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+	"go.mongodb.org/atlas/mongodbatlas"
+)
+
+const (
+	iamEntity             = "iam"
+	orgEntity             = "orgs"
+	projectEntity         = "projects"
+	apiKeysEntity         = "apikeys"
+	apiKeyWhitelistEntity = "whitelist"
+	whitelistIP           = "93.144.26.147"
```
Comment (gssbzn) [inline e2e/iam/helper_test.go:34]: [nit] let's make this random like other whitelist tests

File: e2e/iam/org_api_key_whitelist_test.go [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+			"--force")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		assert.NoError(t, err, string(resp))
+	})
+
+	err := deleteOrgAPIKey(apiKeyID)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
```
Comment (gssbzn) [inline e2e/iam/org_api_key_whitelist_test.go:102]: let's do this like we do with clusters and make it just after creation but with a defer function

File: e2e/iam/project_api_keys_test.go [hunk 5]
```diff
@@ -56,6 +56,19 @@ func TestProjectAPIKeys(t *testing.T) {
...
+		cmd := exec.Command(cliPath, iamEntity,
+			projectEntity,
+			apiKeysEntity,
+			"updates",
+			ID,
+			"--role=GROUP_DATA_ACCESS_READ_ONLY",
+			"-o=json")
+		cmd.Env = os.Environ()
+		resp, err := cmd.CombinedOutput()
+		assert.NoError(t, err, string(resp))
```
Comment (gssbzn) [inline e2e/iam/project_api_keys_test.go:69]: can you please add a deserialisation test as well? check the role you gave it matches?

Comment (andreaangiolillo) [reply e2e/iam/project_api_keys_test.go:69]: this endpoint doesn't return a response. If you want I can run again `list` and check that `update` was successful

Comment (gssbzn) [reply e2e/iam/project_api_keys_test.go:69]: Oh yeah, now I remember, let's let it be then

File: e2e/iam/org_api_key_whitelist_test.go [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+func TestOrgAPIKeyWhitelist(t *testing.T) {
+	cliPath, er := e2e.Bin()
+	if er != nil {
+		t.Fatalf("unexpected error: %v", er)
+	}
+
+	apiKeyID, e := createOrgAPIKey()
+	if e != nil {
+		t.Fatalf("unexpected error: %v", e)
```
Comment (gssbzn) [inline e2e/iam/org_api_key_whitelist_test.go:38]: [nit] 
```suggestion
		t.Fatalf("error deleting test API Key: %v", e)
```

File: e2e/iam/org_api_key_whitelist_test.go [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+		if e := deleteOrgAPIKey(apiKeyID); e != nil {
+			t.Errorf("error deleting test apikey: %v", e)
+		}
+	}()
+
+	n, err := e2e.RandInt(255)
+	if err != nil {
+		t.Fatalf("unexpected error: %v", err)
+	}
+	whitelistIP := fmt.Sprintf("192.168.0.%d", n)
```
Comment (gssbzn) [inline e2e/iam/org_api_key_whitelist_test.go:51]: 👍

File: e2e/iam/project_api_keys_test.go [hunk 8]
```diff
@@ -56,6 +56,19 @@ func TestProjectAPIKeys(t *testing.T) {
 		}
 	})
 
+	t.Run("Assign", func(t *testing.T) {
+		cmd := exec.Command(cliPath, iamEntity,
+			projectEntity,
+			apiKeysEntity,
+			"updates",
```
Comment (gssbzn) [inline e2e/iam/project_api_keys_test.go:63]: do it with `assign`, `updates` is an alias that is not totally correct with the intent of the command

**Pull request merged**